lib/rules/max-lines-per-function.js

```diff
@@ -190,11 +193,24 @@ module.exports = {
 
             if (lineCount > maxLines) {
                 const name = upperCaseFirst(astUtils.getFunctionNameWithKind(funcNode));
+                const linesExceed = lineCount - maxLines;
+                const srcCodeLinesLength = [...sourceCode.lines].pop().length;
+
+                const loc = {
+                    start: {
+                        line: lines[maxLines].lineNumber,
+                        column: 0
+                    },
+                    end: {
+                        line: sourceCode.lines.length,
+                        column: srcCodeLinesLength
+                    }
+                };
 
                 context.report({
-                    node,
```

Comment:
We can keep `node` here, and just add `loc`. If both `node` and `loc` are present, the reported location will be `loc`.

https://eslint.org/docs/developer-guide/working-with-rules#contextreport

`node` will provide additional info (e.g., `nodeType`) for lint messages.